pennylane/measurements/expval.py

```diff
@@ -62,6 +62,12 @@ def circuit(x):
             "qml.expval does not support measuring sequences of measurements or observables"
         )
 
+    if isinstance(op, qml.Identity) and len(op.wires) == 0:
```

Comment:
I tried `op==qml.Identity()` first but that would actually queue `qml.Identity()` 🤯

Reply:
```suggestion
    if len(op.wires) == 0:
```

Potentially even just this?

Reply:
Right!

Reply:
Apparently we support measuring `H = qml.Hamiltonian([], [])` so this doesnt work unfortunately, see [failure1](https://github.com/PennyLaneAI/pennylane/actions/runs/7723559741/job/21053940311?pr=5106) [failure2](https://github.com/PennyLaneAI/pennylane/actions/runs/7723559741/job/21053942479?pr=5106) [failure3](https://github.com/PennyLaneAI/pennylane/actions/runs/7723559741/job/21053941723?pr=5106)

will revert